### PR TITLE
added description 

### PR DESCRIPTION
**Pull request by MaratMussabekov**
issue #3292

### CODE REVIEW

**Comment by e0i** [comment]
@nenonix @JohanFreelancer9 Copy editing is needed for this PR. Thanks.

File: windows/security/threat-protection/auditing/event-5155.md [hunk 1]
```diff
@@ -24,41 +24,123 @@ By default Windows firewall won't prevent a port from being listened by an appli
...
 You can add your own filters using the WFP APIs to block listen to reproduce this event: <https://msdn.microsoft.com/library/aa364046(v=vs.85).aspx>.
 
-There is no event example in this document.
-
 ***Subcategory:***&nbsp;[Audit Filtering Platform Connection](audit-filtering-platform-connection.md)
 
-***Event Schema:***
+***Event Description:***
+
+This event generates every time [Windows Filtering Platform](https://docs.microsoft.com/windows/win32/fwp/windows-filtering-platform-start-page) blocks an application or service from listening on a port for incoming connections.
```
Comment (JohanFreelancer9) [inline windows/security/threat-protection/auditing/event-5155.md]:
```suggestion
This event generates every time the [Windows Filtering Platform](https://docs.microsoft.com/windows/win32/fwp/windows-filtering-platform-start-page) blocks an application or service from listening on a port for incoming connections.
```

File: windows/security/threat-protection/auditing/event-5155.md [hunk 2]
```diff
@@ -24,41 +24,123 @@ By default Windows firewall won't prevent a port from being listened by an appli
...
 
-> *Source Address:%3*
->
-> *Source Port:%4*
->
-> *Protocol:%5*
+**Application Information**:
 
-*Filter Information:*
+-   **Process ID** \[Type = Pointer\]: hexadecimal Process ID of the process which was permitted to bind to the local port. Process ID (PID) is a number used by the operating system to uniquely identify an active process. To see the PID for a specific process you can, for example, use Task Manager (Details tab, PID column):
```
Comment (JohanFreelancer9) [inline windows/security/threat-protection/auditing/event-5155.md]:
```suggestion
-   **Process ID** \[Type = Pointer\]: Hexadecimal Process ID (PID) of the process which was permitted to bind to the local port. PID is a number used by the operating system to uniquely identify an active process. To see the PID for a specific process you can, for example, use Task Manager (Details tab, PID column):
```

File: windows/security/threat-protection/auditing/event-5155.md [hunk 3]
```diff
@@ -24,41 +24,123 @@ By default Windows firewall won't prevent a port from being listened by an appli
...
-***Required Server Roles:*** None.
+    If you convert the hexadecimal value to decimal, you can compare it to the values in Task Manager.
 
-***Minimum OS Version:*** Windows Server 2008, Windows Vista.
+    You can also correlate this process ID with a process ID in other events, for example, “[4688](event-4688.md): A new process has been created” **Process Information\\New Process ID**.
 
-***Event Versions:*** 0.
+<!-- -->
+
+-   **Application Name** \[Type = UnicodeString\]**:** full path and the name of the executable for the process.
```
Comment (JohanFreelancer9) [inline windows/security/threat-protection/auditing/event-5155.md]:
```suggestion
-   **Application Name** \[Type = UnicodeString\]**:** Full path and the name of the executable for the process.
```

File: windows/security/threat-protection/auditing/event-5155.md [hunk 4]
```diff
@@ -24,41 +24,123 @@ By default Windows firewall won't prevent a port from being listened by an appli
...
+
+    Logical disk is displayed in format \\device\\harddiskvolume\#. You can get all local volume numbers by using **diskpart** utility. The command to get volume numbers using diskpart is “**list volume”**:
```
Comment (JohanFreelancer9) [inline windows/security/threat-protection/auditing/event-5155.md]:
```suggestion
    Logical disk is displayed in the format \\device\\harddiskvolume\#. You can get all local volume numbers by using the **diskpart** utility. The command to get volume numbers using diskpart is “**list volume**”:
```

File: windows/security/threat-protection/auditing/event-5155.md [hunk 5]
```diff
@@ -24,41 +24,123 @@ By default Windows firewall won't prevent a port from being listened by an appli
...
+
+<img src="images/diskpart.png" alt="DiskPart illustration" width="786" height="246" />
+
+**Network Information:**
+
+-   **Source Address** \[Type = UnicodeString\]**:** the local IP address of the computer running the application.
```
Comment (JohanFreelancer9) [inline windows/security/threat-protection/auditing/event-5155.md]:
```suggestion
-   **Source Address** \[Type = UnicodeString\]**:** The local IP address of the computer running the application.
```

File: windows/security/threat-protection/auditing/event-5155.md [hunk 6]
```diff
@@ -24,41 +24,123 @@ By default Windows firewall won't prevent a port from being listened by an appli
...
+
+    -   IPv6 Address
+
+    -   :: - all IP addresses in IPv6 format
+
+    -   0.0.0.0 - all IP addresses in IPv4 format
+
+    -   127.0.0.1 , ::1 - localhost
+
+-   **Source Port** \[Type = UnicodeString\]**:** the port number used by the application.
```
Comment (JohanFreelancer9) [inline windows/security/threat-protection/auditing/event-5155.md]:
```suggestion
-   **Source Port** \[Type = UnicodeString\]**:** The port number used by the application.
```

File: windows/security/threat-protection/auditing/event-5155.md [hunk 7]
```diff
@@ -24,41 +24,123 @@ By default Windows firewall won't prevent a port from being listened by an appli
...
+| OSPF Open Shortest Path First                      | 89              |
+| PARC Universal Packet Protocol (PUP)               | 12              |
+| Reliable Datagram Protocol (RDP)                   | 27              |
+| Reservation Protocol (RSVP) QoS                    | 46              |
+
+**Filter Information:**
+
+-   **Filter Run-Time ID** \[Type = UInt64\]: unique filter ID which blocks the application from binding to the port. By default, Windows firewall won't prevent a port from binding by an application, and if this application doesn’t match any filters, you will get value 0 in this field.
+
+    To find specific Windows Filtering Platform filter by ID you need to execute the following command: **netsh wfp show filters**. As a result of this command, **filters.xml** file will be generated. You need to open this file and find the specific substring with the required filter ID (**&lt;filterId&gt;**)**,** for example:
```
Comment (JohanFreelancer9) [inline windows/security/threat-protection/auditing/event-5155.md]:
```suggestion
    To find a specific Windows Filtering Platform filter by ID you need to execute the following command: **netsh wfp show filters**. As a result of this command, a **filters.xml** file will be generated. You need to open this file and find the specific substring with the required filter ID (**&lt;filterId&gt;**), for example:
```

File: windows/security/threat-protection/auditing/event-5155.md [hunk 8]
```diff
@@ -24,41 +24,123 @@ By default Windows firewall won't prevent a port from being listened by an appli
...
+
+    <img src="images/filters-xml-file.png" alt="Filters.xml file illustration" width="840" height="176" />
+
+-   **Layer Name** \[Type = UnicodeString\]: [Application Layer Enforcement](https://msdn.microsoft.com/library/windows/desktop/aa363971(v=vs.85).aspx) layer name.
+
+-   **Layer Run-Time ID** \[Type = UInt64\]: Windows Filtering Platform layer identifier. To find specific Windows Filtering Platform layer ID you need to execute the following command: **netsh wfp show state**. As result of this command **wfpstate.xml** file will be generated. You need to open this file and find specific substring with required layer ID (**&lt;layerId&gt;**)**,** for example:
```
Comment (JohanFreelancer9) [inline windows/security/threat-protection/auditing/event-5155.md]:
```suggestion
-   **Layer Run-Time ID** \[Type = UInt64\]: Windows Filtering Platform layer identifier. To find a specific Windows Filtering Platform layer ID, you need to execute the following command: **netsh wfp show state**. As result of this command, a **wfpstate.xml** file will be generated. You need to open this file and find the specific substring with the required layer ID (**&lt;layerId&gt;**), for example:
```

File: windows/security/threat-protection/auditing/event-5155.md [hunk 9]
```diff
@@ -24,41 +24,123 @@ By default Windows firewall won't prevent a port from being listened by an appli
...
+| Internet Group Management Protocol (IGMP)          | 88              |
+| MIT Remote Virtual Disk (RVD)                      | 66              |
+| OSPF Open Shortest Path First                      | 89              |
+| PARC Universal Packet Protocol (PUP)               | 12              |
+| Reliable Datagram Protocol (RDP)                   | 27              |
+| Reservation Protocol (RSVP) QoS                    | 46              |
+
+**Filter Information:**
+
+-   **Filter Run-Time ID** \[Type = UInt64\]: unique filter ID which blocks the application from binding to the port. By default, Windows firewall won't prevent a port from binding by an application, and if this application doesn’t match any filters, you will get value 0 in this field.
```
Comment (JohanFreelancer9) [inline windows/security/threat-protection/auditing/event-5155.md]:
```suggestion
-   **Filter Run-Time ID** \[Type = UInt64\]: A unique filter ID which blocks the application from binding to the port. By default, Windows firewall won't prevent a port from binding to an application, and if this application doesn’t match any filters, you will get a 0 value in this field.
```

File: windows/security/threat-protection/auditing/event-5155.md [hunk 10]
```diff
@@ -24,41 +24,123 @@ By default Windows firewall won't prevent a port from being listened by an appli
...
 
-***Event Schema:***
+***Event Description:***
+
+This event generates every time [Windows Filtering Platform](https://docs.microsoft.com/windows/win32/fwp/windows-filtering-platform-start-page) blocks an application or service from listening on a port for incoming connections.
+
+<br clear="all">
+
+***Event XML:***
+```
```
Comment (illfated) [inline windows/security/threat-protection/auditing/event-5155.md]:
```suggestion
```xml
```

File: windows/security/threat-protection/auditing/event-5155.md [hunk 11]
```diff
@@ -24,41 +24,123 @@ By default Windows firewall won't prevent a port from being listened by an appli
...
+        <Data Name="Application">\device\harddiskvolume2\users\test\desktop\netcat\nc.exe</Data>
+        <Data Name="SourceAddress">0.0.0.0</Data>
+        <Data Name="SourcePort">5555</Data>
+        <Data Name="Protocol">6</Data>
+        <Data Name="FilterRTID">84576</Data>
+        <Data Name="LayerName">%%14609</Data>
+        <Data Name="LayerRTID">40</Data>
+    </EventData>
+</Event>
+
```
Comment (illfated) [inline windows/security/threat-protection/auditing/event-5155.md]:
```suggestion
```

**Comment by e0i** [comment]
@MaratMussabekov Please follow up with copy editing suggestions. Thanks.

**Comment by MaratMussabekov** [comment]
Dear @JohanFreelancer9, @illfated,
your recommendations were applied,
thank you!

**Comment by e0i** [comment]
@Dansimp This PR introduces a description and an XML for Windows Event Log _Event 5155_ and has passed copy review. Please review and merge as the time permits.

Issue reference: #3292

Thanks

**Comment by e0i** [comment]
@Dansimp Following up for updates. Thanks.

**Comment by e0i** [comment]
@Dansimp Following up for updates.

**Pull request merged**